File: lib/path_loss.js
/*!
 * Path Loss
 * Copyright(c) 2019 Hüseyin Yiğit
 * MIT Licensed
 */

/**
 * Path Loss.
 * @param {number} freq - Frequency in MHz.
 * @param {number} dist - Propagation distance in km.
 * @return {number} Path loss in dB.
 */
module.exports = function(freq, dist) {
    return 32.44 + 20 * Math.log10(freq * dist);
}

File: src/pathloss.cpp
#include <exception>
#include <math.h>
#include "pathloss.hpp"

float softroles::propagation::pathloss(float f, float d) {
  try {
    return 32.44 + 20*std::log10(f*d);
  }
  catch(std::exception& e) {
    throw;
  }
}
